File: server/test/sc/server/plugins/TestPlayer.java
package sc.server.plugins;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import sc.framework.plugins.Player;
import sc.api.plugins.ITeam;

public class TestPlayer extends Player {
  private static Logger logger = LoggerFactory.getLogger(TestPlayer.class);

  public TestPlayer(ITeam<?> pc) {
    super(pc);
  }

  public void requestMove() {
    TestTurnRequest request = new TestTurnRequest();
    logger.info(toString() + " is requesting a move");
    notifyListeners(request);
  }

}

File: server/test/sc/server/plugins/TestPlayer.kt
package sc.server.plugins

import org.slf4j.LoggerFactory
import sc.api.plugins.ITeam
import sc.framework.plugins.Player

class TestPlayer(pc: ITeam) : Player(pc) {
    override fun requestMove() {
        val request = TestTurnRequest()
        logger.info("$this is requesting a move")
        notifyListeners(request)
    }

    companion object {
        private val logger = LoggerFactory.getLogger(TestPlayer::class.java)
    }
}